### PR TITLE
Feature/check user status

### PR DESCRIPTION
**Pull request by damola-sd**
# Description

Added endpoints to 
-Check if a user is a mentor 
-Check if a user is a mentee
-Save Mentoring choices for mentors and mentees

## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

File: api/controllers/mentor.js [hunk 1]
```diff
@@ -66,5 +66,44 @@ module.exports = {
...
+          where: { user_id: userId },
+          attributes: ['id', 'location_id', 'industry_id']
+        });
+        if (mentor) {
+          const newMentor = { ...mentor.dataValues, mentor: true };
+          return response.success(res, 200, newMentor);
+        }
+        return response.success(res, 201, 'user is not a mentor');
+      }
+      return response.error(res, 500, 'user is not found');
```
Comment (iamcristos) [inline api/controllers/mentor.js]: why is this a 500

File: api/controllers/mentor.js [hunk 2]
```diff
@@ -66,5 +66,44 @@ module.exports = {
...
+    try {
+      const { mentorTypeId, mentorId } = req.body;
+      const mentorChoice = await models.Mentors_choices.create({
+        mentoring_type_id: mentorTypeId,
+        mentor_id: mentorId
+      });
+      if (mentorChoice) return response.success(res, 201, mentorChoice);
+      return response.error(res, 404, 'could assign mentoring type to mentor');
+    } catch (error) {
+      return response.error(res, 404, error.message);
```
Comment (iamcristos) [inline api/controllers/mentor.js]: why is this a 404

File: api/validator/userValidator.js [hunk 3]
```diff
@@ -53,9 +53,7 @@ module.exports = {
   validateUserProfileUpdate(req, res, next) {
     const validator = new Validator(req.body, {
       firstName: 'required|alpha',
-      lastName: 'required|alpha',
-      countryName: 'required|alpha',
-      cityName: 'required|alpha'
+      lastName: 'required|alpha'
     });
```
Comment (iamcristos) [inline api/validator/userValidator.js]: for the location endpoint getting the id is there a validation that validates the country name and city name???

File: api/controllers/mentee.js [hunk 4]
```diff
@@ -66,5 +66,44 @@ module.exports = {
     } catch (error) {
       return response.error(res, 500, error.message);
     }
+  },
+
+  async checkifUserIsMentee(req, res) {
```
Comment (ruihildt) [inline api/controllers/mentee.js]: Don't forget to camelCase the `If`.

File: api/controllers/mentee.js [hunk 5]
```diff
@@ -66,5 +66,44 @@ module.exports = {
...
+
+  async addMenteeChoice(req, res) {
+    try {
+      const { mentorTypeId, menteeId } = req.body;
+      const menteeChoice = await models.Mentees_choices.create({
+        mentoring_type_id: mentorTypeId,
+        mentee_id: menteeId
+      });
+      if (menteeChoice) return response.success(res, 201, menteeChoice);
+      return response.error(res, 404, 'Could assign mentoring type to mentee');
```
Comment (ruihildt) [inline api/controllers/mentee.js]: Update the error message with negation `Could not assign mentorint type to mentee`.

File: api/validator/userValidator.js (hunk 3 above)
Comment (iykekings) [reply api/validator/userValidator.js]: It is sent from the autocomplete API to the front end...so the front end just returns the right city and country. But we can still have validation on the backend.

**Pull request merged**